### PR TITLE
Careers pages redesign

### PR DESCRIPTION
**Pull request by reemhamz**
## One-line summary
We're updating the careers pages for a newer and fresher look!

New pages will be added, as well as an updated, modern design with updated information to match current work life at Mozilla.

## Issue / Bugzilla link
- https://github.com/mozilla/bedrock/issues/11143
- RSS feed adder: https://github.com/mozilla/bedrock/issues/12422

## Checklist

If relevant:

- [x] Add new pages and to subnav
- [x] Route new pages within `views.py` and `urls.py`
- [x] Remove old Protocol file in directory and `@import` it to stylesheets
- [x] Remove internships URL and no-index it (usused page and will be redesigned later in 2023)
- [x] Add content

## Testing

Demo server URL: (or None)

https://www-demo1.allizom.org/en-US/careers/

To test this work:

-  http://localhost:8000/en-US/careers/
-  http://localhost:8000/en-US/careers/diversity/
-  http://localhost:8000/en-US/careers/teams/
-  http://localhost:8000/en-US/careers/locations/
-  http://localhost:8000/en-US/careers/benefits/

### CODE REVIEW

File: media/css/careers/modules/overview.scss
Comment (nathan-barrett) [inline media/css/careers/modules/overview.scss:106]: Wonder if these could be replaced with `:nth-child(even)` and `:nth-child(odd)`

File: media/js/careers/testimonials-modal.es6.js
Comment (nathan-barrett) [inline media/js/careers/testimonials-modal.es6.js:111]: **Question** I _think_ that this should be `articleArray.length -1`? 

Ex: 
```js
let arr = [1, 2, 4, 5, 6];
console.log(arr.length);
-> 5
console.log(arr[arr.length]);
-> undefined
console.log(arr[arr.length -1]);
-> 6
```

File: media/static-bundles.json
Comment (nathan-barrett) [inline media/static-bundles.json:1200]: See comment in styles.scss

File: media/css/careers/styles.scss
Comment (nathan-barrett) [inline media/css/careers/styles.scss:15]: I couple **Nits** here:
- I don't think that every page in careers needs to use all of these files, right? If they don't, I think we should should separate them into different bundles. (For instance, listings.html will need `job-listings.scss` but not `diversity.scss`) Would be interested to hear why that might be a good idea for performance, but it would make sense to my brain
- I think we should rename the `module` folder to `components`since that is a pattern that we use in other places in bedrock, but `modules` is a naming convention that we don't use in other places

**Review by nathan-barrett — APPROVED** [review]
Way to kick butt @reemhamz! I had a bunch of nitpicks and questions but nothing that I saw that should block this from being merged. Killer job getting this all together, it looks so good!

File: bedrock/careers/templates/careers/diversity.html
Comment (reemhamz) [reply bedrock/careers/templates/careers/diversity.html:18]: Nice catch!

File: bedrock/careers/templates/careers/home.html
Comment (reemhamz) [reply bedrock/careers/templates/careers/home.html:332]: From my understanding, since the video is embedded onto our site, we're adding `nocookie` so that way no cookies are traced when the video is played directly on our website. However, the anchor tag is a fallback incase the `iframe` doesn't work and takes the user to the official Youtube page for the video to play. 
cc: @craigcook @alexgibson, happy for y'alls input in case I missed anything with the explanation

Comment (alexgibson) [reply bedrock/careers/templates/careers/home.html:332]: We use this pattern elsewhere so I think this is fine as-is. If someone blocks iframes then they might not entertain watching youtube either, but this is probably the best we can do in this instance.

File: bedrock/careers/templates/careers/benefits.html
Comment (reemhamz) [reply bedrock/careers/templates/careers/benefits.html:41]: So regarding this, I tried multiple different things in terms of alignment:
1. Instead of using `mzp-t-columns-three`, I used `mzp-t-columns-two`:
Screen Shot 2022-11-23 at 11 31 44 AM



2. I played around with the grid layout and moved the last item to be more towards the right:
Screen Shot 2022-11-23 at 11 32 34 AM


3. I used `display: flex` instead of `display: grid` and played around with the widths of the list items:
Screen Shot 2022-11-23 at 11 31 06 AM


I didn't personally like any of the solutions I tried out and liked the original layout better. What are your thoughts?

File: bedrock/careers/templates/careers/includes/big-picture-video.html
Comment (reemhamz) [reply bedrock/careers/templates/careers/includes/big-picture-video.html:9]: TIL!

File: media/js/careers/testimonials-modal.es6.js
Comment (reemhamz) [reply media/js/careers/testimonials-modal.es6.js:111]: I don't think so. If we place `-1` at the end of that if-statement, then it skips over the last item in that array, because instead of counting up to `5` it is just counting up to `4`!

File: media/css/careers/styles.scss
Comment (reemhamz) [reply media/css/careers/styles.scss:15]: I _personally_ think it's fine that we're structuring it this way? We usually code out our stylesheets in this manner most of the time?

File: bedrock/careers/templates/careers/benefits.html
Comment (nathan-barrett) [reply bedrock/careers/templates/careers/benefits.html:40]: eh.. I don't think there is a good solution here. Thanks for checking it out!

**Review by nathan-barrett — APPROVED** [review]

**Comment by nathan-barrett** [comment]
Needs a rebase before a merge, but otherwise looks good!

**Comment by reemhamz** [comment]
There have been some design changes. I updated most of what was needed and did a small code clean-up. Would appreciate a re-review when all design changes are confirmed @nathan-barrett. I'll let you know when it's ready

File: bedrock/careers/templates/careers/includes/footer.html
Comment (alexgibson) [inline bedrock/careers/templates/careers/includes/footer.html:18]: Drive by nit: (was just browsing the demo site, nice job 👋): These blue `mzp-t-product` themed buttons are typically used for product links, like "Download Firefox". For consistency, could this alternatively be a regular black Mozilla themed link?

(Feel free to ignore, this is not a hill to die on, but i think consistency throughout the site can be important to consider).

File: bedrock/careers/templates/careers/base.html
Comment (nathan-barrett) [inline bedrock/careers/templates/careers/base.html:43]: **Nit**: Inconsistent capitalization here

**Review by nathan-barrett — APPROVED** [review]
arrrrplus

File: bedrock/careers/templates/careers/base.html
Comment (reemhamz) [reply bedrock/careers/templates/careers/base.html:43]: The pattern of titles on our navs usually have each word capitalized, so that's why I capitalized it

Comment (nathan-barrett) [reply bedrock/careers/templates/careers/base.html:43]: oh, i'm dumb - I was looking at the cta-name not the text. Can forget about this

**Pull request merged**
